blockmanager_test.go

```diff
@@ -89,168 +297,46 @@ func TestBlockManagerInitialInterval(t *testing.T) {
 			t.Fatal(err)
 		}
 
-		var blockHeaders []headerfs.BlockHeader
-		blockHeaders = append(blockHeaders, headerfs.BlockHeader{
-			BlockHeader: genesisBlockHeader,
-			Height:      0,
-		})
-
 		genesisFilterHeader, _, err := cfStore.ChainTip()
 		if err != nil {
 			t.Fatal(err)
 		}
 
-		var cfHeaders []headerfs.FilterHeader
-		cfHeaders = append(cfHeaders, headerfs.FilterHeader{
-			HeaderHash: genesisBlockHeader.BlockHash(),
-			FilterHash: *genesisFilterHeader,
-			Height:     0,
-		})
-
-		// The filter hashes (not the filter headers!) will be sent as
-		// part of the CFHeaders response, so we also keep track of
-		// them
-		genesisFilter, err := builder.BuildBasicFilter(
-			chaincfg.SimNetParams.GenesisBlock, nil,
-		)
-		if err != nil {
-			t.Fatalf("unable to build genesis filter: %v", err)
-		}
-
-		genesisFilterHash, err := builder.GetFilterHash(genesisFilter)
+		headers, err := generateHeaders(genesisBlockHeader,
+			genesisFilterHeader, func(*chainhash.Hash) {})
```

Comment:
Seems we can just pass `nil` here instead of this funky function literal.

Reply:
With `generateHeaders` being modified to check for nil-ness.

Reply:
Fixed.